script/run-comp-match

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3
+"""
+A script to run a competition match.
+"""
+
+import os
+import sys
+import shutil
+import argparse
+import tempfile
+import contextlib
+import subprocess
+from typing import Iterator
+from pathlib import Path
+from zipfile import ZipFile
+
+REPO_ROOT = Path(__file__).resolve().parent.parent
+
+sys.path.insert(1, str(REPO_ROOT / 'modules'))
+
+import controller_utils  # isort:skip
+
+
+def get_zone_path(zone_id: int) -> Path:
+    robot_file: Path = controller_utils.get_zone_robot_file_path(zone_id)
+    return robot_file.parent
+
+
+def construct_match_data(args: argparse.Namespace) -> controller_utils.MatchData:
+    tlas = [
+        tla if tla != '-' else None
+        for tla in args.tla
+    ]
+
+    return controller_utils.MatchData(
+        args.match_num,
+        tlas,
+        args.duration,
+        recording_config=None,  # use the default
+    )
+
+
+@contextlib.contextmanager
+def temporary_arena_root(suffix: str) -> Iterator[None]:
+    arena_root = controller_utils.ARENA_ROOT
```

Comment:
Is there a reason we are backing up and restoring `ARENA_ROOT` given that we never use the old value

Reply:
Mostly so that this context manager is easily re-usable in the event that it gets copied/moved for re-use.